sklearn/manifold/_barnes_hut_tsne.pyx

```diff
@@ -0,0 +1,625 @@
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: cdivision=True
+# Author: Christopher Moody <chrisemoody@gmail.com>
+# Author: Nick Travers <nickt@squareup.com>
+import numpy as np
+from cython.parallel cimport prange, parallel
+from libc.stdio cimport printf
+cimport numpy as np
+cimport cython
+
+# Implementation by Chris Moody & Nick Travers
+# original code by Laurens van der Maaten
+# for reference implemenations and papers describing the technique:
+# http://homepage.tudelft.nl/19j49/t-SNE.html
+
+
+from libc.stdlib cimport malloc, free, abs
+
+cdef extern from "math.h":
+    double sqrt(double x) nogil
+
+
+# This is effectively an ifdef statement in Cython
+# It allows us to write printf debugging lines
+# and remove them at compile time
+cdef enum:
+    DEBUGFLAG = 0
```

Comment:
Does this actually work?

If I remove entirely:
```python
cdef enum:
    DEBUGFLAG = 0
```

And add in setup.py:

```python
...extra_compile_args=["-O3", "-DDEBUGFLAG=1"])...
```
Then Cython complains that DEBUGFLAG isn't a declared variable (`Accessing Python global or builtin not allowed without gil`) -- which of course it isn't. 

So then declaring the DEBUGFLAG but leaving it uninitialized (`def int DEBUGFLAG`) compiles, but doesn't seem to respect the compile args (`-DDEBUGFLAG=1`). Only if I initialize `cdef int DEBUGFLAG=1` does it work. 

Would love guidance here. I've left this using:
```python
cdef enum:
    DEBUGFLAG = 0
```
for the time being.